collect_logs.py

```diff
@@ -0,0 +1,188 @@
+    try:
+        stdin, stdout, stderr = ssh_install.exec_command("sudo ceph --version")
+        if not stderr.read().decode():
+            file_name = "cephansible.sh"
+        else:
+            file_name = "cephadm.sh"
+        file = os.path.join(os.path.dirname(__file__), f"{file_name}")
+        sftp = ssh_install.open_sftp()
+        sftp.put(f"{file}", f"/home/{uname}/{file}")
+        ssh_install.exec_command(
+            f"sudo sh {file} > ceph_status.txt; sudo mv ceph_status.txt {log_dir}"
+        )
+    except Exception:
+        results[nodeip] = False
+
+
+def generate_sosreport_in_node(
+    ssh_install, nodeip: str, uname: str, pword: str, log_dir: str, results: dict
+) -> None:
+    """Generate sosreport in the given node and copy report to installer
+
+    Args:
+       ssh_install         ssh object of installer node
+       nodeip              host Ip address
+       uname               Username for accessing host
+       pword               password for accessing host through given user
+       log_dir             directory to store all the logs
+       results             host Ip address with boolean parameters
+
+    Returns:
+        None
+    """
+    print(f"Connecting {nodeip} to generate sosreport")
+    try:
+        results[nodeip] = True
+        ssh_d = SSHConnectionManager(nodeip, uname, pword).get_client()
+        ssh_d.exec_command("sudo yum -y install sos")
+        stdin, stdout, stderr = ssh_d.exec_command(
+            "sudo sosreport -a --all-logs -e ceph --batch"
+        )
+        sosreport = re.search(r"/var/tmp/sosreport-.*.tar.xz", stdout.read().decode())
+        print(f"Successfully generated sosreport in node {nodeip} :{sosreport.group()}")
+        ssh_d.exec_command(f"sudo chown {uname}:{uname} {sosreport.group()}")
+        ssh_install.exec_command(f"scp {nodeip}:{sosreport.group()} {log_dir}")
+        ssh_d.exec_command(f"sudo rm -rf {sosreport.group()}")
+        ssh_d.close()
+        print(f"Successfully copied report to {nodeip}")
+    except Exception:
+        results[nodeip] = False
+
+
+def run(args: dict) -> int:
+    """Standard script to collect all the logs from ceph cluster through installer node
+
+    Through installer node get all other nodes in the cluster, generate sosreport for all the nodes obtained.
+    Run shell script on installer node, then upload all the collected logs to magna
+
+    Args:
+       ip         installer IP address
+       username   username to be used to access the system other than root
+       password   password for installer node
+
+    Returns:
+        0 on success or 1 for failures
+
+    Raises:
+        AssertionError: An error occurred if given IP is not of Installer node
+    """
+    results = dict()
+    run_id = generate_unique_id(length=6)
+    ip = args["--ip"]
+    uname = args["--username"]
+    pword = args["--password"]
+    log_dir = f"ceph_logs_{run_id}"
+    file_share = "http://magna002.ceph.redhat.com/cephci-jenkins"
```

Comment:
I was thinking that we should move the logs location under `ceph-jenkins/ceph-<run_id>/ceph_logs`

Reply:
Changed Location to cephci-jenkins/ceph_logs_{run_id}/logs